Solution1/ConsoleApp/Services/PerformerCommandsService.cs

```diff
@@ -55,5 +57,47 @@ public Task CloseApplication()
             Console.WriteLine(Constants.Notifications.CloseApp);
             return Task.CompletedTask;
         }
+
+        public async Task GetBestWeatherAsync()
+        {
+            Console.WriteLine("Please, enter array city name (separator symbal - ',') :");
+            var stringCityNames = Console.ReadLine();
+            if (string.IsNullOrEmpty(stringCityNames))
+            {
+                Console.WriteLine(Constants.Validation.IncorrectValue);
+                _logger.LogError($"User entered incorrect value for 'countDay'.");
+                return;
+            }
+
+            var listCityNames = stringCityNames.Split(',').Select(cityName => cityName.Trim());
+            var weatherResponsesDTO = await _weatherServiсe.GetWeatherByArrayCityNameAsync(listCityNames);
+            var bestWeather = weatherResponsesDTO.Where(w => w.IsSuccessfulRequest).OrderByDescending(w => w.Temp).FirstOrDefault();
+
+            if (bestWeather != null)
+            {
+                Console.WriteLine($"City with the highest temperature {bestWeather.Temp} C: {bestWeather.CityName}.");
+            }
+
+            var dictionaryWeatherResponsesDTO = weatherResponsesDTO.GroupBy(w => w.IsSuccessfulRequest).ToDictionary(k => k.Key, v => v.ToList());
+            int countSuccessRequest = dictionaryWeatherResponsesDTO.TryGetValue(true, out var successRequest) ? successRequest.Count() : 0;
+            int countFailRequest = dictionaryWeatherResponsesDTO.TryGetValue(false, out var failRequest) ? failRequest.Count() : 0;
+
+            Console.WriteLine
```

Comment:
read your Task again. You have incorrect output logic now, both for city and for common result

Reply:
+